ui/Rhino/AGS/dev/AGS_form_fix_nodes_cmd.py

```diff
@@ -24,8 +24,16 @@ def RunCommand(is_interactive):
         vertices = form.select_vertices()
         if not vertices:
             break
-        for vertex in vertices:
-            form.diagram.vertex_attribute(vertex, 'is_fixed', True)
+        current = str(form.diagram.vertex_attribute(vertices[0], 'is_fixed'))  # display T/F based on first selected
+        show = compas_rhino.rs.GetString("Fix Nodes", defaultString=current, strings=["True", "False"])
+        if show == "True":
+            for vertex in vertices:
+                form.diagram.vertex_attribute(vertex, 'is_fixed', True)
+        elif show == "False":
+            for vertex in vertices:
+                form.diagram.vertex_attribute(vertex, 'is_fixed', False)
+        else:
+            pass
```

Comment:
```
    # select fixed vertices
    while True:
        vertices = form.select_vertices()
        if not vertices:
            break
        for vertex in vertices:
            if form.diagram.vertex_attribute(vertex, 'is_fixed') is True:
                form.diagram.vertex_attribute(vertex, 'is_fixed', False)
            else:
                form.diagram.vertex_attribute(vertex, 'is_fixed', True)
        scene.update()
```
I tried this way. it works... do you think it's better?

Reply:
Your code works much faster indeed @duchaoyu. It may just be not so intuitive for someone that never used it.

If @brgcode prefers it your way I would suggest renaming the command to `AGS_form_(un)fix_nodes` so the user has a hint that he can also unselect a node by selecting an already-fixed node.

Reply:
```
    # select fixed vertices
    while True:
        vertices = form.select_vertices()
        if not vertices:
            break
        fixed = []
        unfixed = []
        for vertex in vertices:
            if form.diagram.vertex_attribute(vertex, 'is_fixed') is True:
                form.diagram.vertex_attribute(vertex, 'is_fixed', False)
                unfixed.append(vertex)
            else:
                form.diagram.vertex_attribute(vertex, 'is_fixed', True)
                fixed.append(vertex)
        scene.update()
        if fixed and unfixed: 
            print(fixed, 'nodes are fixed,', unfixed, 'nodes are unfixed') 
        if fixed and not unfixed:
            print(fixed, 'nodes are fixed')
        if unfixed and not fixed:
            print(unfixed, 'nodes are unfixed'
```
This is better.

Reply:
I am not a big fan of the print text. I think the update on the scene speaks by itself since the the color on the vertices will change based on True/False for fixity.